exporters/provider_common/openshift.py

```diff
@@ -19,3 +69,234 @@ def convert_datetime(dt: Union[str, datetime]) -> datetime:
+        if replica:
+            return {owner_ref.uid: replica}
+
+        logging.debug(
+            "Getting replica: %s, kind: %s, api_version: %s",
+            owner_ref.uid,
+            owner_ref.kind,
+            owner_ref.apiVersion,
+        )
+
+        try:
+            api_resource = client.resources.get(
+                api_version=owner_ref.apiVersion, kind=owner_ref.kind
+            )
+
+            # We don't need to limit for a given namespace, because Replica objects may live in other
+            # then Pod namespace, so we may get multiple replica objects for a given name.
+            # The field_selector does not work on the UID, that's why we need to match separately
+            replica_list = api_resource.get(
+                field_selector=f"metadata.name={owner_ref.name}"
+            )
+
+            for replica in replica_list.items:
+                if replica.metadata.uid == owner_ref.uid:
+                    _add_object_to_cache(owner_ref.uid, replica)
+                    return {owner_ref.uid: replica}
+        except ResourceNotFoundError:
+            logging.debug(
+                "API Object not found for version: %s object: %s",
+                owner_ref.apiVersion,
+                owner_ref.uid,
+            )
+    return {}
+
+
+def filter_pods_by_replica_uid(
+    pods_list: List[ResourceField],
+) -> Dict[str, ResourceField]:
+    """
+    Filters out the given list of Pod objects to create a dictionary with ReplicaSet or
+    ReplicationController UIDs as keys.
+
+    In OpenShift, one Pod may have multiple ownerReferences. This function filters the list of
+    Pod objects to identify unique ReplicaSet or ReplicationController objects based on their UID,
+    which is stored in the ownerReferences attribute of the Pod object.
+
+    Since Pods within a ReplicaSet are homogeneous and identical replicas,
+    we can filter out unique ReplicaSet objects even if multiple Pods are deployed.
+
+    Args:
+        pods_list (List[ResourceField]): A list of Pod objects.
+
+    Returns:
+        Dict[str, ResourceField]
+                A dictionary with ReplicaSet or ReplicationController UIDs as keys and Pod objects as values.
+    """
+    uids = set()
+    pods = {}
+
+    for pod in pods_list:
+        for owner_reference in pod.metadata.ownerReferences or []:
+            replica_uid = owner_reference.uid
+            if replica_uid in uids:
+                continue
+            uids.add(replica_uid)
+            pods[replica_uid] = pod
+
+    return pods
+
+
+def get_and_log_namespaces(
+    client: DynamicClient, namespaces: set[str], prod_label: str
+) -> set[str]:
+    """
+    Get the set of namespaces to watch, and log what they are.
+    They will be either:
+    1. The namespaces explicitly specified
+    2. The namespaces matched by PROD_LABEL
+    3. If neither namespaces nor the PROD_LABEL is given, then implicitly matches all namespaces.
+    """
+    if namespaces:
```

Comment:
this function was not touched, just moved here, right? But I think we can improve it

here shouldn't we check if the namespaces exist first? 

and in line 252 (log.info) should be a else from the following if

Reply:
No, it's perfectly valid scenario to setup exporter first and then create that namespace later.